Guide/file-storage.markdown

```diff
@@ -575,3 +575,248 @@ config = do
+
+        -- Get the original image directory and UUID from the path.
+        let (originalImageDirectory, uuid) = extractDirectoryAndUUID originalImagePath
+
+
+        let imageStylePathDirectory = originalImageDirectory <> "/imageStyles/" <> size
+        let imageStylePath = imageStylePathDirectory <> "/" <> uuid
+
+        -- If we use a StaticDirStorage storage then we need to prefix the path with the `static/` folder.
+        let storagePrefix = case storage of
+                StaticDirStorage -> "static/"
+                _ -> ""
+
+        fileExists <- doesFileExist (cs $ storagePrefix <> imageStylePath)
+
+        if fileExists
+            then do
+                -- Image style found.
+                renderFile (cs $ storagePrefix <> imageStylePath) "application/jpg"
+            else do
+                -- Image style not found, so create it.
+                let options :: StoreFileOptions = def
+                        { directory = imageStylePathDirectory
+                        , preprocess = applyImageMagick "jpg" ["-resize", cs size <> "^", "-gravity", "center", "-extent", cs size, "-quality", "85%", "-strip"]
+                        }
+
+                storedFile <- storeFileFromPath (cs $ storagePrefix <> originalImageDirectory <> "/" <> uuid) options
+
+                renderFile (cs $ storagePrefix <> storedFile.path) "application/jpg"
+
+-- | Extracts the directory and UUID from a path like "pictures/8ed22caa-11ea-4c45-a05e-91a51e72558d"
+extractDirectoryAndUUID :: (?context :: context, ConfigProvider context) => Text -> (Text, Text)
+extractDirectoryAndUUID inputText =
+    case reverse parts of
+        uuid : pathSegments -> (T.intercalate "/" (reverse pathSegments), uuid)
+        _ -> ("", "")
+    where
+        frameworkConfig = ?context.frameworkConfig
+        trimmedText = T.replace (frameworkConfig.baseUrl <> "/") "" inputText
+        parts = T.splitOn "/" trimmedText
+```
+
+We need to change `RenderImageStyleAction` to get another argument, the signed token.
+
+
+```haskell
+-- Web/Types.hs
+-- ...
+data ImageStyleController
+    = = RenderImageStyleAction { width :: !Int, height :: !Int, originalImagePath :: !Text, signed :: !Text }
+    deriving (Eq, Show, Data)
+```
+
+Now, our `Show` action is responsible for generating the signed token. We can do it like this:
+
+```haskell
+[hsx|<img src={pathTo $ RenderImageStyleAction 400 200 imageUrl signed} />|]
+where
+    imageUrl = "http://localhost:8000/static/picture/b4c8f55c-16d6-41f0-9503-77352b134e14"
+
+    -- Sign the image URL to prevent tampering.
+    Config.RsaPublicAndPrivateKeys (_, privateKey) = getAppConfig @Config.RsaPublicAndPrivateKeys
+    signed = case RSA.sign Nothing (Just Hash.Algorithms.SHA256) privateKey (cs $ imageUrl <> "400x200") of
+        Left msg -> error $ "Cannot sign image URL, private key is invalid:" <> show msg
+        Right signature -> signature |> Base64.encode |> cs
```

Comment:
```suggestion
        -- Base 64 decode the token so that it can be used in the URL.
        Right signature -> signature |> Base64.encode |> cs
```